### ISSUE TITLE
Refactor to a class

### ISSUE THREAD

**Issue by targos** [mljs/kmeans#12]
I'd like to refactor the library to a class, with an API similar to sklearn: http://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html

I'll migrate to TypeScript at the same time.

@mljs/collaborators WDYT?

**Comment by maasencioh** [comment]
why TypeScript? I think that the class is not a bad idea, do you see any advantage doing this?

**Comment by stropitek** [comment]
I think that's a very good idea. People will be familiar with the API. I think most people doing ML have worked with python and scikit before.

@maasencioh we've been experimenting with Typescript and it's awesome.
- Auto-completion and documentation as you code
- Fix typos and type errors before even running the program
- Thinking with types in mind helps you to design a better API IMHO
- We otten have a build step anyway (for example to support es6 modules)

**Comment by jobo322** [comment]
Syntax is like Java, for reading the code maybe it's better, you need to write more code for the same result but personally I like the comments functionality for the argument types

**Comment by lpatiny** [comment]
Migrating to a class would be great indeed but I would prefer to have something similar to pca, pls and other packages

```js
const kMeans = new KMeans(dataset, options); (or new KMeans(dataset, numberClusters, options))

const newPoints = [[4.9, 3.2, 1.2, 0.4], [5.4, 3.3, 1.4, 0.9]];
console.log(kMeans.predict(newPoints)); // project new points into the PCA space

const model2 = kMeans.toJSON(); // to save the model

const kMeans2 = KMeans.load(model);
kMeans2.predict(newPoints);
```
